vendor/github.com/ciscoecosystem/mso-go-client/client/client_service.go

```diff
@@ -96,10 +96,12 @@ func (c *Client) DeletebyId(url string) error {
 	if err1 != nil {
 		return err1
 	}
-	if resp.StatusCode == 204 || resp.StatusCode == 200 {
-		return nil
-	} else {
-		return fmt.Errorf("Unable to delete the object")
+	if resp != nil {
```

Comment:
Why have you included the block in :if resp != nil?

Reply:
Because when statusCode == 204, c.Do does return nil, nil, nil so you get a nil error when trying to access resp.StatusCode. I preferred to wrap it in an if statement than changing the whole logic to avoid to much changes.